### ISSUE TITLE
Argument of type 'O' is not assignable to parameter of type 'RoundedRectangle'.

### ISSUE THREAD

**Issue by RSUmitcan** [amcharts/amcharts5#312]
Hi,

I was using amcharts5 in Angular.
I copied "Column with Rotated Label" source code [https://www.amcharts.com/demos/column-with-rotated-series/](url).

But in 
```
series.columns.template.adapters.add("fill", (fill, target) => {
  return chart.get("colors").getIndex(series.columns.indexOf(target));
});

series.columns.template.adapters.add("stroke", (stroke, target) => {
  return chart.get("colors").getIndex(series.columns.indexOf(target));
});
```

part it throws me an error.

```Argument of type 'O' is not assignable to parameter of type 'RoundedRectangle'.
  Type 'IRoundedRectangleSettings' is missing the following properties from type 'RoundedRectangle': _settings, _privateSettings, _beforeChanged, _draw, and 191 more
```

when i delete this part column colors doesn't change. I think documentation should be fixed.

**Comment by martynasma** [comment]
Unless you want to use `any`, here's a stronger type:

```JavaScript
series.columns.template.adapters.add("fill", function(fill, target) {
  return chart.get("colors")!.getIndex(series.columns.indexOf(target as unknown as am5.RoundedRectangle));
});

series.columns.template.adapters.add("stroke", function(stroke, target) {
  return chart.get("colors")!.getIndex(series.columns.indexOf(target as unknown as am5.RoundedRectangle));
});
```

I'm afraid we won't be able to modify the demo code as that is a generic code, including ES6 which does not support generics.

**Comment by Pauan** [comment]
We were able to improve the types for adapters, so in the next release everything should work perfectly without any errors.

**Comment by martynasma** [comment]
Fixed in [5.1.8](https://github.com/amcharts/amcharts5/releases/tag/5.1.8).

## [5.1.8] - 2022-03-10

### Added
- Settings `pinchZoomX` and `pinchZoomY` (defaults: `false`) added to `XYChart`. If set to `true` will allow pinch-zooming of a chart on touch devices. NOTE: these settings are not supported on `RadarChart`.
- `topPlotContainer` added to `XYChart`. It's a `Container` that can be used to place element over plot area.

### Changed
- The types for adapters has been changed to be more intuitive and useful. ([Issue 312](https://github.com/amcharts/amcharts5/issues/312))
- Tapping an element on a touch device will now generate `pointerover` event in addition to `click`. When tapped somewhere else, `pointerout` event will be triggered.

### Fixed
- Fixing ES6 module detection by BundlePhobia again ([Issue 294](https://github.com/amcharts/amcharts5/issues/294)).
- Pinch-zooming of a `MapChart` was not working well in some cases.
- `maxHeight` setting in `Sprite` was not working.
- Change of `minWidth` was not being applied at once, only if some other changes triggered chart to be redrawn.
- Changing or not setting data on an existing `FlowChart` was resulting in a JS error.
- `maxPanOut` setting in `MapChart` was not working.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
